From 8dea3ec1361823a2d52e76dd5591b321bd8778a9 Mon Sep 17 00:00:00 2001
From: Archfx <cybx@live.com>
Date: Fri, 29 Nov 2024 18:15:44 -0500
Subject: [PATCH] Fix add drawer margins during rotation

---
 .../allapps/AllAppsTransitionController.java  | 24 ++++++++++++++++++
 1 files changed, 24 insertions(+)

diff --git a/src/com/android/launcher3/allapps/AllAppsTransitionController.java b/src/com/android/launcher3/allapps/AllAppsTransitionController.java
index 1b0ad04e89..22ce5faabe 100644
--- a/src/com/android/launcher3/allapps/AllAppsTransitionController.java
+++ b/src/com/android/launcher3/allapps/AllAppsTransitionController.java
@@ -67,6 +67,14 @@ import com.android.launcher3.util.ScrollableLayoutManager;
 import com.android.launcher3.util.Themes;
 import com.android.launcher3.util.VibratorWrapper;
 import com.android.launcher3.views.ScrimView;
+
+import com.android.launcher3.allapps.ActivityAllAppsContainerView;
+import com.android.launcher3.R;
+import android.widget.FrameLayout;
+import android.view.ViewGroup;
+import android.content.Context;
+import android.widget.RelativeLayout;
 
 /**
  * Handles AllApps view transition.
@@ -221,6 +229,23 @@ public class AllAppsTransitionController
         }
 
         mIsTablet = dp.isTablet;
+        if (mLauncher.getStateManager().getState() == ALL_APPS) {
+            resetAllAppsMargins();
+        }
+    }
+
+    private void resetAllAppsMargins() {
+        
+        ActivityAllAppsContainerView<Launcher> mAppsView = mLauncher.findViewById(R.id.apps_view);
+        View mBottomSheetBackground = mLauncher.findViewById(R.id.bottom_sheet_background);
+        FrameLayout.LayoutParams params = (FrameLayout.LayoutParams) mAppsView.getLayoutParams();
+        RelativeLayout.LayoutParams backgroundParams = (RelativeLayout.LayoutParams) mBottomSheetBackground.getLayoutParams();
+        params.width = ViewGroup.LayoutParams.MATCH_PARENT; // Full width
+        params.height = ViewGroup.LayoutParams.MATCH_PARENT; // Full height
+        backgroundParams.leftMargin = 0;
+        backgroundParams.rightMargin = 0;
+        mAppsView.setLayoutParams(params);
+        mBottomSheetBackground.setLayoutParams(backgroundParams);
     }
 
     /**
--
2.34.1

